diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98ec960d..54f609b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,6 +68,9 @@ if(UNIX AND NOT DIST_INSTALL)
     endif()
 endif()
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+
 add_subdirectory(garglk)
 
 # xBRZ requires C++17.
diff --git a/garglk/CMakeLists.txt b/garglk/CMakeLists.txt
index f14d55d0..9b58e47a 100644
--- a/garglk/CMakeLists.txt
+++ b/garglk/CMakeLists.txt
@@ -122,7 +122,7 @@ endif()
 find_package(Freetype REQUIRED)
 find_package(PNG 1.6 REQUIRED)
 target_include_directories(garglk PUBLIC cheapglk PRIVATE ${FREETYPE_INCLUDE_DIRS} ${PNG_INCLUDE_DIRS})
-target_link_libraries(garglk PRIVATE ${FREETYPE_LIBRARIES} ${PNG_LIBRARIES})
+target_link_libraries(garglk PRIVATE ${FREETYPE_LIBRARIES} ${PNG_LIBRARIES} pthread)
 
 if(NOT WITH_BUNDLED_FMT)
     find_package(fmt CONFIG)
@@ -155,7 +155,7 @@ set_target_properties(garglkmain PROPERTIES POSITION_INDEPENDENT_CODE "${GARGLK_
 if(WITH_LAUNCHER)
     add_executable(gargoyle WIN32 launcher.cpp)
     target_include_directories(gargoyle PRIVATE cheapglk)
-    target_link_libraries(gargoyle PRIVATE garglk)
+    target_link_libraries(gargoyle PRIVATE garglk pthread)
     add_fmt(gargoyle)
     cxx_standard(gargoyle ${CXX_VERSION})
     warnings(gargoyle)
diff --git a/terps/CMakeLists.txt b/terps/CMakeLists.txt
index 86a9c271..60ea61df 100644
--- a/terps/CMakeLists.txt
+++ b/terps/CMakeLists.txt
@@ -24,7 +24,7 @@ function(terp target)
     set_target_properties(${target} PROPERTIES OUTPUT_NAME "${GARGLKPRE}${target}")
     target_compile_definitions(${target} PRIVATE GARGLK ${TERP_MACROS})
     target_include_directories(${target} PRIVATE ${TERP_INCLUDE_DIRS})
-    target_link_libraries(${target} PRIVATE garglkmain garglk ${TERP_LIBS})
+    target_link_libraries(${target} PRIVATE garglkmain garglk ${TERP_LIBS} pthread)
 
     if(MINGW OR MSVC)
         target_sources(${target} PRIVATE "${PROJECT_SOURCE_DIR}/garglk/icons.rc" "${PROJECT_SOURCE_DIR}/terps/garglk.manifest")
